### PR TITLE
Add and delete points from a line. 

### PR DESCRIPTION
**Pull request by JeffThorslund**
Holding the "Control" key and clicking...

-  in space on the SVG will create a point on the last selected line.
- on a point will delete that point. 

Its a bit messy, but its a good start.

### CODE REVIEW

File: client/src/components/shared/Point.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React from "react";
+import { useKeyPress } from "../shared/_utils";
```
Comment (bekatru) [inline client/src/components/shared/Point.js:2]: Can you tell me how this works? 
You are importing a component from a folder with this syntax?
Does underscore mean something in `_utils`?

File: client/src/rematch/models.js [hunk 2]
```diff
@@ -258,5 +258,43 @@ export const testEnvironment = {
       state.draggedCubic = false;
       return state;
     },
+    addPoint: (state, payload) => {
+      const { coords } = payload;
+      const { activeLine } = state;
+      const vectorLength = state.lines[activeLine].vector.length - 1;
```
Comment (bekatru) [inline client/src/rematch/models.js:264]: I am confused. Why are you substracting the length by 1?

**Review by bekatru — COMMENTED** [review]
Have you tried this on eddys? From what I see it should work on those as well!

File: client/src/components/shared/Point.js (hunk 1 above)
Comment (JeffThorslund) [reply client/src/components/shared/Point.js:2]: If we need somewhere to store hooks, or any other special functions, we can store them in `_utils`, short for "utilities". Files in `_utils` can be used in any of the components in the parent folder.

File: client/src/rematch/models.js (hunk 2 above)
Comment (JeffThorslund) [reply client/src/rematch/models.js:264]: Good catch, that was the wrong variable name. It should have been something like `lastPointIndex`. It is used when finding the coordinates of the last point. 

Ill fix that.

**Comment by JeffThorslund** [comment]
> Have you tried this on eddys? From what I see it should work on those as well!

It doesnt quite work on eddys in its current form, but shouldnt take too much extra work!

**Pull request merged**